frappe/public/js/frappe/router.js

```diff
@@ -302,8 +302,10 @@ frappe.router = {
 			if (route[2] && route[2] !== 'list' && !$.isPlainObject(route[2])) {
 				new_route = [this.slug(route[1]), 'view', route[2].toLowerCase()];
 
-				// calendar / inbox / file folder
-				if (route[3]) new_route.push([...route].slice(3, route.length));
+				if (route[3]) {
+						// calendar / inbox / file folder
```

Comment:
I think this should be before / next to the condition, since it clarifies the condition rather than the block itself.